staging/src/k8s.io/apiserver/pkg/util/webhook/client.go

```diff
@@ -142,7 +143,9 @@ func (cm *ClientManager) HookClient(cc ClientConfig) (*rest.RESTClient, error) {
 		// Use http/1.1 instead of http/2.
 		// This is a workaround for http/2-enabled clients not load-balancing concurrent requests to multiple backends.
 		// See https://issue.k8s.io/75791 for details.
-		cfg.NextProtos = []string{"http/1.1"}
+		if isLocalHost(cc.URL) {
```

Comment:
Both service and URL-based versions go through this code path. We also are already parsing the URL below.

This would be clearer as a `forceHTTP1 := true` which we *only* set to false after parsing `u, err := url.Parse(cc.URL)` below and checking the hostname there for localhost / loopback

Reply:
Done, thanks!

Reply:
hmm, @aojea you suggested what I initially did but @liggitt suggested otherwise. I have no strong opinion on either.

Reply:
I think @aojea was suggesting moving the NextProtos assignment down below to where we had already parsed the URL (do it unconditionally in the service branch, and conditionally in the URL branch)

Reply:
I don't feel this is simpler since we still need to change it in two places.

Reply:
I may be missing something, why do we have such large diff?  do we need to breaking it down in `func (cm *ClientManager) hookClientConfig(cc ClientConfig) (*rest.Config, error) {`

Reply:
> I may be missing something, why do we have such large diff? 

it was due to bringing several dependencies. I removed them since your suggestion doesn't need them.

> do we need to breaking it down in func (cm *ClientManager) hookClientConfig(cc ClientConfig) (*rest.Config, error) {

yes, this refactoring would facilitate writing the unit test https://github.com/linxiulei/kubernetes/commit/612c36df69ef846d517f319b4e79acadfcfd86eb#diff-0afdfbb98099d9c00bfb801579318a05e3f226eb384d50561cef6886e7631046R1